Gems/Atom/RPI/Code/Tests/Material/MaterialSourceDataTests.cpp

```diff
@@ -965,7 +1015,239 @@ namespace UnitTest
+            {
+                "propertyLayout": {
+                    "propertyGroups":
+                    [
+                        {
+                            "name": "general",
+                            "properties": [
+                                {
+                                    "name": "MyFloat",
+                                    "type": "Float"
+                                },
+                                {
+                                    "name": "MyFloat2",
+                                    "type": "Vector2"
+                                },
+                                {
+                                    "name": "MyColor",
+                                    "type": "Color"
+                                }
+                            ]
+                        }
+                    ]
+                }
+            }
+        )";
+
+        AZ::Utils::WriteFile(simpleMaterialTypeJson, "@exefolder@/Temp/test.materialtype");
+
+        const AZStd::string material1Json = R"(
+            {
+                "materialType": "@exefolder@/Temp/test.materialtype",
+                "propertyValues": {
+                    "general.MyFloat": 1.5,
+                    "general.MyColor": [0.1, 0.2, 0.3, 0.4]
+                }
+            }
+        )";
+        
+        AZ::Utils::WriteFile(material1Json, "@exefolder@/Temp/m1.material");
+        
+        const AZStd::string material2Json = R"(
+            {
+                "materialType": "@exefolder@/Temp/test.materialtype",
+                "parentMaterial": "@exefolder@/Temp/m1.material",
+                "propertyValues": {
+                    "general.MyFloat2": [4.1, 4.2],
+                    "general.MyColor": [0.15, 0.25, 0.35, 0.45]
+                }
+            }
+        )";
+        
+        AZ::Utils::WriteFile(material2Json, "@exefolder@/Temp/m2.material");
+        
+        const AZStd::string material3Json = R"(
+            {
+                "materialType": "@exefolder@/Temp/test.materialtype",
+                "parentMaterial": "@exefolder@/Temp/m2.material",
+                "propertyValues": {
+                    "general.MyFloat": 3.5
+                }
+            }
+        )";
+        
+        AZ::Utils::WriteFile(material3Json, "@exefolder@/Temp/m3.material");
+        
+        MaterialSourceData sourceDataLevel1 = MaterialUtils::LoadMaterialSourceData("@exefolder@/Temp/m1.material").TakeValue();
+        MaterialSourceData sourceDataLevel2 = MaterialUtils::LoadMaterialSourceData("@exefolder@/Temp/m2.material").TakeValue();
+        MaterialSourceData sourceDataLevel3 = MaterialUtils::LoadMaterialSourceData("@exefolder@/Temp/m3.material").TakeValue();
+        
+        auto materialAssetLevel1 = sourceDataLevel1.CreateMaterialAssetFromSourceData(Uuid::CreateRandom());
+        EXPECT_TRUE(materialAssetLevel1.IsSuccess());
+        EXPECT_TRUE(materialAssetLevel1.GetValue()->WasPreFinalized());
```

Comment:
Careful, if the previous check fails and it doesn't return successful, then `GetValue` could return `nullptr`, causing this test run to crash. EXPECT_* checks do not stop the test from running, you have to protect all the pointers in case the check fails.

````
EXPECT_TRUE(materialAssetLevel3.IsSuccess());
if (materialAssetLevel3.IsSuccess())
{
    EXPECT_TRUE(materialAssetLevel3.GetValue()->WasPreFinalized());
}
````

Same for `layout` and other `GetValue()` calls in this test.